### PR TITLE
[BIOMAGE-2252] - Add new subset pipeline to subset experiments

### PR DESCRIPTION
**Pull request by gerbeldo**
# Description
Implements a new gem2s step to subset experiments.

# Details
#### URL to issue
https://biomage.atlassian.net/browse/BIOMAGE-2252

#### Link to staging deployment URL (or set N/A)
https://ui-martinfosco-ui138-api78-pi.scp-staging.biomage.net/

#### Links to any PRs or resources related to this PR
https://github.com/biomage-org/api/pull/78

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/R/subset-1-subset_seurat.R [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+#'
+#' Loads the processed rds and cellsets file from the parent experiment from s3.
+#'
+#' @param input list of input parameters
+#' @param pipelne_config list of pipeline parameters
+#'
+#' @return list with scdata and parsed cellsets
+#' @export
+#'
+load_parental_data <- function(input, pipeline_config) {
```
Comment (cosa65) [inline pipeline-runner/R/subset-1-subset_seurat.R:65]: I think this should be called `load_parent_experiment` or `load_parent_experiment_data`, parental data is a little bit weird

File: pipeline-runner/R/gem2s-6-prepare_experiment.R [hunk 2]
```diff
@@ -15,7 +14,7 @@
 prepare_experiment <- function(input, pipeline_config, prev_out) {
   message("Preparing experiment ...")
 
-  check_names <- c("config", "counts_list", "annot", "doublet_scores", "scdata_list", "disable_qc_filters")
+  check_names <- c("config", "edrops", "annot", "scdata_list", "disable_qc_filters")
```
Comment (saracastel) [inline pipeline-runner/R/gem2s-6-prepare_experiment.R:17]: Why is the this changed? We now have `edrops` that wasn't there before, and we don't have `doublet_scores` anymore. These seems unrelated to the subsetting

File: pipeline-runner/R/subset-1-subset_seurat.R [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+#'   - subsetExperimentId character
+#'   - cellSetKeys character vector of cellset keys to subset
+#'   - experimentName character
+#' @param pipeline_config list
+#' @param prev_out list, ignored because this is the first step in the subset pipeline
+#'
+#' @return list containing scdata_list, annotations and sample_id_map
+#' @export
+#'
+create_subset_experiment <- function(input, pipeline_config, prev_out = NULL) {
```
Comment (saracastel) [inline pipeline-runner/R/subset-1-subset_seurat.R:18]: Is `prev_out` needed in this function? It seems it's not used anywhere

**Review by saracastel — APPROVED** [review]

File: pipeline-runner/R/subset-1-subset_seurat.R (hunk 3 above)
Comment (gerbeldo) [reply pipeline-runner/R/subset-1-subset_seurat.R:18]: every gem2s and pipeline step has prev_out as input, even if they do not use it. Check `download_user_files` for example which is the absolute first step. This has to do with the way the init functions are implemented. After discussing with martin, `NULL` is a better choice than `list()` (used elsewhere), because it's actually not used.

File: pipeline-runner/R/gem2s-6-prepare_experiment.R (hunk 2 above)
Comment (gerbeldo) [reply pipeline-runner/R/gem2s-6-prepare_experiment.R:17]: it is kind of related. the check names vector contains the things that must be present for the step to work. I suppose it was copypasted between steps, because it always checks that the same things are there, and not the actual needed elements. for example `edrops`. It is used to decide whether or not to enable the classifier filter. `doublet_scores` on the other hand, are not used at all in this nor the following steps.

File: pipeline-runner/R/subset-1-subset_seurat.R (hunk 1 above)
Comment (gerbeldo) [reply pipeline-runner/R/subset-1-subset_seurat.R:65]: if you're a mask you're a mask

Comment (gerbeldo) [reply pipeline-runner/R/subset-1-subset_seurat.R]: but changed, since I agree with your maskiness

File: pipeline-runner/R/gem2s-6-prepare_experiment.R (hunk 2 above)
Comment (gerbeldo) [reply pipeline-runner/R/gem2s-6-prepare_experiment.R:17]: we will need to change this when we implement the per-sample classifier filter.
